src/Event.php

```diff
@@ -636,6 +652,10 @@ public function toArray(): array
             $data['contexts']['runtime'] = $this->runtimeContext->toArray();
         }
 
+        if (!$this->contexts->isEmpty()) {
```

Comment:
This has to be changed.
We want to merge with existing `$data['contexts']` in the event but the one set from the user should be stronger.
Please also provide tests for it since this is a common use case.

Reply:
Done :)